### PR TITLE
[permissions] vendor don't see every metadata descriptors

### PR DESCRIPTION
**Pull request by EvanBldy**
**Problem**
- vendors see every metadata descriptors and every data information

**Solution**
- vendor don't see every metadata descriptors, they see only metadata descriptors with no departments or metadata descriptors corresponding to their departments, the same for data in entity

### CODE REVIEW

**Comment by frankrousseau** [comment]
It breaks, many tests. It requires some fixes.

**Comment by EvanBldy** [comment]
> It breaks, many tests. It requires some fixes.

It should be fixed in the next tests :)

File: zou/app/blueprints/assets/resources.py [hunk 1]
```diff
@@ -139,6 +139,9 @@ def get(self):
             criterions["assigned_to"] = persons_service.get_current_user()[
                 "id"
             ]
+            criterions[
```
Comment (frankrousseau) [inline zou/app/blueprints/assets/resources.py:142]: It should not be added if the user is not a vendor.

File: zou/app/services/assets_service.py [hunk 2]
```diff
@@ -258,6 +259,15 @@ def get_assets_and_tasks(criterions={}, page=1, with_episode_ids=False):
                 str(link.entity_in_id)
             )
 
+    query_result = tasks_query.all()
+
+    if "vendor_departments" in criterions:
+        not_allowed_descriptors_field_names = entities_service.get_not_allowed_descriptors_fields_for_entity_type_and_departments(
```
Comment (frankrousseau) [inline zou/app/services/assets_service.py:265]: * The function name is too long
* The line limit width is 80 chars

File: zou/app/blueprints/assets/resources.py (hunk 1 above)
Comment (EvanBldy) [reply zou/app/blueprints/assets/resources.py:142]: It's the case there's permissions.has_vendor_permissions() line 138

File: zou/app/services/assets_service.py (hunk 2 above)
Comment (EvanBldy) [reply zou/app/services/assets_service.py]: It's fixed

**Comment by frankrousseau** [comment]
It seems good. But it requires to add some tests to ensure the expected behaviour is respected.

**Comment by EvanBldy** [comment]
> It seems good. But it requires to add some tests to ensure the expected behaviour is respected.

I will add them next week :)

**Comment by EvanBldy** [comment]
@frankrousseau the tests are added, I think it's ready to be merged :)

**Pull request merged**